### PR TITLE
Add actor picker component to new settlement reports page.

### PR DESCRIPTION
**Pull request by FirestarJes**

### CODE REVIEW

File: apps/dh/api-dh/source/DataHub.WebApi/GraphQL/GraphQLQuery.cs [hunk 1]
```diff
@@ -134,8 +138,8 @@ public GraphQLQuery()
 
             Field<NonNullGraphType<ListGraphType<NonNullGraphType<BatchType>>>>("batches")
                 .Argument<DateRangeType>("executionTime")
-                .Argument<BatchState>("executionState", nullable: true)
-                .Argument<ProcessType>("processType", nullable: true)
+                .Argument<BatchState>("executionState", true)
```
Comment (mimse) [inline apps/dh/api-dh/source/DataHub.WebApi/GraphQL/GraphQLQuery.cs:141]: Suggestion: keep named param, so it is easier to see what is set to true

File: apps/dh/api-dh/source/DataHub.WebApi/GraphQL/GraphQLQuery.cs [hunk 2]
```diff
@@ -163,7 +167,7 @@ public GraphQLQuery()
 
             Field<NonNullGraphType<ListGraphType<NonNullGraphType<SettlementReportType>>>>("settlementReports")
                 .Argument<ProcessTypeEnum>("processType")
-                .Argument<string[]>("gridAreaCodes", nullable: true)
+                .Argument<string[]>("gridAreaCodes", true)
```
Comment (mimse) [inline apps/dh/api-dh/source/DataHub.WebApi/GraphQL/GraphQLQuery.cs:170]: Suggestion: keep named param, so it is easier to see what is set to true

File: apps/dh/api-dh/source/DataHub.WebApi/GraphQL/GraphQLQuery.cs [hunk 3]
```diff
@@ -36,7 +36,8 @@ public GraphQLQuery()
                 .Resolve()
                 .WithScope()
                 .WithService<IMarketParticipantPermissionsClient>()
-                .ResolveAsync(async (context, client) => await client.GetPermissionAsync(context.GetArgument<int>("id")));
+                .ResolveAsync(
```
Comment (mimse) [inline apps/dh/api-dh/source/DataHub.WebApi/GraphQL/GraphQLQuery.cs:39]: Question: Looks like your formatter is different that the standard one.

**Review by mimse — CHANGES_REQUESTED** [review]

File: apps/dh/api-dh/source/DataHub.WebApi/GraphQL/GraphQLQuery.cs (hunk 3 above)
Comment (FirestarJes) [reply apps/dh/api-dh/source/DataHub.WebApi/GraphQL/GraphQLQuery.cs:39]: hmm, have to double check, but it should just be the default Rider one

File: apps/dh/api-dh/source/DataHub.WebApi/GraphQL/GraphQLQuery.cs (hunk 2 above)
Comment (FirestarJes) [reply apps/dh/api-dh/source/DataHub.WebApi/GraphQL/GraphQLQuery.cs:170]: Will do

File: apps/dh/api-dh/source/DataHub.WebApi/GraphQL/GraphQLQuery.cs (hunk 1 above)
Comment (FirestarJes) [reply apps/dh/api-dh/source/DataHub.WebApi/GraphQL/GraphQLQuery.cs]: fixed

**Review by mimse — DISMISSED** [review]

**Pull request merged**